### PR TITLE
Adds ability to intersect EntityTypes

### PR DESCRIPTION
**Pull request by csilvestrini**

### CODE REVIEW

File: java/arcs/core/analysis/EntityTypeOperations.kt [hunk 1]
```diff
@@ -34,6 +39,19 @@ infix fun Schema.union(other: Schema): Outcome<Schema> {
     return Schema(names = newNames, fields = newFields, hash = "").toSuccess()
 }
 
+/**
+ * Computes the intersection of the two [Schema] instances. Returns [Outcome.Failure] if the union
```
Comment (bgogul) [inline java/arcs/core/analysis/EntityTypeOperations.kt:43]: The comment is not consistent with the implementation. There can be failures when we try to intersect types, right?

Comment (csilvestrini) [reply java/arcs/core/analysis/EntityTypeOperations.kt:43]: Thanks, fixed comment.

There's actually no way for this code to fail. If the types of a field don't match, we just ignore it. See https://github.com/PolymerLabs/arcs/blob/master/src/runtime/schema.ts#L158

Comment (bgogul) [reply java/arcs/core/analysis/EntityTypeOperations.kt]: That does not sound right. Suppose that the fields have types `&Person {name}` and `&Person{name, age}`. Don't we want it to be `&Person{name}`?

Comment (bgogul) [reply java/arcs/core/analysis/EntityTypeOperations.kt]: Let us file a bug for now? We can take care of this when we implement the type inference in Kotlin?

Comment (csilvestrini) [reply java/arcs/core/analysis/EntityTypeOperations.kt]: Filed b/156983624

**Pull request merged**